structured/structured.go

```diff
@@ -226,3 +226,22 @@ func (c *ColumnType) SQLString() string {
 	}
 	return c.Kind.String()
 }
+
+// Validate validates that the database descriptor is well formed.
+// Checks include validate the database name, and verifying that there
+// is at least one read and write user.
+func (desc *DatabaseDescriptor) Validate() error {
+	if err := validateName(desc.Name, "table"); err != nil {
+		return err
+	}
+	if desc.ID == 0 {
```

Comment:
a lot of it can, both here and in plan.go. Refactoring is coming later in this PR.